app/routes.php

```diff
@@ -95,10 +97,16 @@ function( Request $request ) use ( $ffFactory ) {
 		] );
 
 		$responseModel = $useCase->addSubscription( $subscriptionRequest );
-		// TODO forward/dispatch to matching 'page/name' route, depending on $responseModel->getType();
-		return 'TODO';
+		if ( in_array( 'application/json', $request->getAcceptableContentTypes() ) ) {
+			return $app->json( $ffFactory->newAddSubscriptionJSONPresenter()->present( $responseModel ) );
+		}
+		if ( $responseModel->isSuccessful() ) {
+			return $app->redirect( $app['url_generator']->generate('page', [ 'pageName' => 'SubscriptionSuccess' ] ) );
```

Comment:
This is intentional: When the user is forwarded after submitting a form, he can hit the reload button without accidentally resubmitting it. See https://en.wikipedia.org/wiki/Post/Redirect/Get
The redirect is only for the success state, the error state will present the form again.